### ISSUE TITLE
[0.4.5] restrict-plus-operands loses any types through object destructuring with allowAny:false

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1273]
## Summary

With explicit `allowAny:false`, an any value obtained via object destructuring is not reported by `@typescript-eslint/restrict-plus-operands`. The minimal case does not involve callbacks or chaining:

```tsx
export function example(input: any) {
  const { value } = input;
  return value + 1;
}
```

ESLint reports the any operand; utlint emits no diagnostics. Destructured any receivers also cause `.map()` callback index additions to be missed. This happens both when the parent is any and when the parent has an explicitly any-typed property, including a destructured default array.

The original directly annotated receiver examples in #1271 now pass in 0.4.5. This is a separate remaining destructuring boundary.

Verified with @utoo/lint 0.4.5, ESLint 8.57.1, @typescript-eslint/parser and eslint-plugin 6.21.0, Node.js 22.23.1 on macOS arm64. Explicit allowAny:false avoids relying on version-specific defaults. Only this rule is enabled, without parsing errors or suppression comments.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.5 eslint@8.57.1 @typescript-eslint/parser@6.21.0 @typescript-eslint/eslint-plugin@6.21.0 typescript@5.3.3
```

`tsconfig.json`:

```json
{"compilerOptions":{"strict":true,"target":"ES2022","jsx":"react"},"include":["example.tsx"]}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "project": "./tsconfig.json"
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
}
```

Run each snippet separately as `example.tsx`:

```sh
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
```

## destructured-any-plus

```tsx
export function example(input:any){const {value}=input;return value+1;}
```

ESLint: 1 diagnostic(s). utlint: 0 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## destructured-any-direct

```tsx
export function example(input:any){const {items}=input;return items.map((value,index)=>index+1);}
```

ESLint: 1 diagnostic(s). utlint: 0 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## destructured-any-property

```tsx
export function example(input:{items:any}){const {items}=input;return items.map((value,index)=>index+1);}
```

ESLint: 1 diagnostic(s). utlint: 0 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## destructured-array-default

```tsx
export function example(input:any){const {items=[]}=input;return items.map((value,index)=>index+1);}
```

ESLint: 1 diagnostic(s). utlint: 0 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## chain-fallback

```tsx
export function example(input:any){const {items}=input||{};return items.slice(0).map((value,index)=>index+1);}
```

ESLint: 1 diagnostic(s). utlint: 0 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## member-any-map

```tsx
export function example(input:any){return input.items.map((value,index)=>index+1);}
```

ESLint: 1 diagnostic(s). utlint: 1 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## aliased-any-map

```tsx
export function example(input:any){const items=input.items;return items.map((value,index)=>index+1);}
```

ESLint: 1 diagnostic(s). utlint: 1 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## destructured-typed-array

```tsx
export function example(input:{items:number[]}){const {items}=input;return items.slice(0).map((value,index)=>index+1);}
```

ESLint: 0 diagnostic(s). utlint: 0 diagnostic(s).

Expected: preserve any through destructuring and report these invalid additions when allowAny:false. Keep the typed number[] control clean.
